client/landing/gutenboarding/onboarding-block/acquire-intent/vertical-select/index.tsx

```diff
@@ -59,109 +57,129 @@ const VerticalSelect: React.FunctionComponent< Props > = ( { onSubmit } ) => {
-		setInputValue( vertical.label );
 		setIsFocused( false ); // prevent executing handleBlur()
-		onSubmit();
 	};
 
-	const handleSuggestionChangeEvent = ( e: React.ChangeEvent< HTMLInputElement > ) => {
-		setInputValue( e.target.value );
+	const handleBlur = () => {
+		const lastQuery = inputText.trim();
+		if ( isFocused && lastQuery.length ) {
+			const vertical = suggestions.find( ( { label } ) =>
+				label.toLowerCase().includes( lastQuery )
+			) ?? { label: lastQuery, id: '', slug: '' };
+			handleSelect( vertical );
+		}
 	};
 
-	const handleInputKeyDownEvent = ( e: React.KeyboardEvent< HTMLInputElement > ) => {
+	const handleInputKeyDownEvent = ( e: React.KeyboardEvent< HTMLSpanElement > ) => {
+		const input = e.currentTarget.innerText.trim();
+
+		if ( suggestionRef.current ) {
+			suggestionRef.current.handleKeyEvent( e );
+		}
 		if ( e.keyCode === ENTER ) {
 			e.preventDefault();
-			handleSelect( { label: inputValue } );
+			input.length && ! suggestions.length && handleSelect( { label: input } );
 			return;
 		}
-		if ( suggestionRef.current ) {
-			suggestionRef.current.handleKeyEvent( e );
-		}
-	};
-
-	const handleBlur = () => {
-		if ( isFocused ) {
-			const vertical = suggestions.find( ( { label } ) =>
-				label.toLowerCase().includes( normalizedInputValue )
-			) ?? { label: inputValue.trim() };
-
-			handleSelect( vertical );
+		if ( e.keyCode === TAB ) {
+			e.preventDefault();
+			handleBlur();
 		}
 	};
 
 	React.useEffect( () => {
-		if ( ! hasValue ) {
-			inputRef?.current?.focus();
+		if ( isInputEmpty ) {
+			inputRef.current.focus();
 		}
 	}, [] ); // eslint-disable-line react-hooks/exhaustive-deps
 
+	React.useEffect( () => {
+		inputRef.current.innerText = siteVertical?.label || '';
+	}, [ siteVertical, inputRef ] );
+
 	// TODO: Write a better translators comment.
 	// translators: Form input for a site's topic where "<Input />" is replaced by user input and must be preserved verbatim in translated string.
 	const madlibTemplate = NO__( 'My site is about <Input />' );
 	const madlib = __experimentalCreateInterpolateElement( madlibTemplate, {
 		Input: (
 			<span className="vertical-select__suggestions-wrapper">
-				<input
-					ref={ inputRef }
-					/* eslint-disable-next-line wpcalypso/jsx-classname-namespace */
-					className="madlib__input"
-					autoComplete="off"
-					style={ {
-						width: `${ inputValue.length * 0.85 }ch`,
-					} }
-					onBlur={ handleBlur }
-					onFocus={ () => setIsFocused( true ) }
-					onChange={ handleSuggestionChangeEvent }
-					onKeyDown={ handleInputKeyDownEvent }
-					placeholder={ animatedPlaceholder }
-					value={ inputValue }
-				/>
+				<span className="vertical-select__input-wrapper">
+					{ isInputEmpty && (
+						<span className="vertical-select__placeholder">{ animatedPlaceholder }</span>
+					) }
+					<span
+						contentEditable
+						tabIndex={ 0 }
+						role="textbox"
+						aria-multiline="true"
+						spellCheck={ false }
```

Comment:
Same issue with Grammarly.

Reply:
same as https://github.com/Automattic/wp-calypso/pull/40645#discussion_r402183562 we'll wait for the design decision on this